### PR TITLE
Consume unlocked, non-namespaced releases

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

# Issues Closed

- We fixed an issue causing CumulusCI to fail to install releases that contain an Unlocked Package without a namespace.

### CODE REVIEW

**Comment by davidmreed** [comment]
Builds on https://github.com/SFDO-Tooling/CumulusCI/pull/2832

**Comment by davisagli** [comment]
@davidmreed This needs a conflict resolved

File: cumulusci/core/dependencies/dependencies.py [hunk 1]
```diff
@@ -329,8 +329,14 @@ def flatten(self, context: BaseProjectConfig) -> List[Dependency]:
             )
         )
 
-        # Deploy the project, if unmanaged.
-        if not managed:
+        if not self.managed_dependency:
```
Comment (davisagli) [inline cumulusci/core/dependencies/dependencies.py:332]: Would it be reasonable to rename `managed_dependency` to `packaged_dependency`?

Comment (davidmreed) [reply cumulusci/core/dependencies/dependencies.py]: Sure. Done.

**Review by bethbrains — APPROVED** [review]

**Pull request merged**